clients/drcachesim/tracer/raw2trace.cpp

```diff
@@ -2282,10 +2362,17 @@ raw2trace_t::unread_last_entry(raw2trace_thread_data_t *tdata)
     VPRINT(5, "Unreading last entry\n");
     if (tdata->last_entry_is_split) {
         VPRINT(4, "Unreading both parts of split entry at once\n");
-        tdata->pre_read.push_back(tdata->last_split_first_entry);
+        tdata->pre_read.push_front(tdata->last_split_first_entry);
         tdata->last_entry_is_split = false;
     }
-    tdata->pre_read.push_back(tdata->last_entry);
+    tdata->pre_read.push_front(tdata->last_entry);
+}
+
+void
+raw2trace_t::unread_entry(raw2trace_thread_data_t *tdata, offline_entry_t &entry)
+{
+    VPRINT(5, "Unreading a given entry type=%d\n", static_cast<int>(entry.addr.type));
+    tdata->pre_read.push_back(entry);
```

Comment:
It's a little confusing that we do a push_back here but a push_front above, and call both of those operations "Unreading". Maybe call this one queue_entry?

Reply:
Done.